syntax = 'proto3';
package model_server;

service ModelServer {
    rpc predict (GrpcPredictRequest) returns (GrpcPredictResponse) {}
}

message XRow {
    float f0 = 1;
    float f1 = 2;
    float f2 = 3;
    float f3 = 4;
    float f4 = 5;
}

message GrpcPredictRequest {
    int32 ClientId = 1;
    repeated XRow X = 2;
    string ModelName = 3;
}

message GrpcPredictResponse {
    float Prob = 1;
}
